interop/xds/server/server.go

```diff
@@ -78,8 +93,76 @@ func (s *testServiceImpl) EmptyCall(ctx context.Context, _ *testpb.Empty) (*test
 }
 
 func (s *testServiceImpl) UnaryCall(ctx context.Context, in *testpb.SimpleRequest) (*testpb.SimpleResponse, error) {
+	response := &testpb.SimpleResponse{ServerId: s.serverID, Hostname: s.hostname}
+
+	for _, headerVal := range getRPCBehaviorMetadata(ctx) {
+		if strings.HasPrefix(headerVal, sleepPfx) {
+			sleep, err := strconv.Atoi(headerVal[len(sleepPfx):])
+			if err != nil {
+				return response, status.Errorf(codes.InvalidArgument, "invalid format for rpc-behavior header: %v", headerVal)
+			}
+
+			time.Sleep(time.Duration(sleep) * time.Second)
+
+		} else if strings.HasPrefix(headerVal, keepOpenVal) {
+			return nil, nil
+
+		} else if strings.HasPrefix(headerVal, errorCodePfx) {
+			errCd, err := strconv.Atoi(headerVal[len(errorCodePfx):])
+			if err != nil {
+				return response, status.Errorf(codes.InvalidArgument, "invalid format for rpc-behavior header: %v", headerVal)
+			}
+
+			return response, status.Errorf(codes.Code(errCd), "rpc failed as per the rpc-behavior header value: %v", headerVal)
+
+		} else if strings.HasPrefix(headerVal, successOnRetryPfx) {
+			attempt, err := strconv.Atoi(headerVal[len(successOnRetryPfx):])
+			if err != nil {
+				return response, status.Errorf(codes.InvalidArgument, "invalid format for rpc-behavior header: %v", headerVal)
+			}
+
+			mdRetry := getMetadataValues(ctx, rpcAttemptsMdKey)
+			retry, err := strconv.Atoi(mdRetry[0])
+			if err != nil {
+				return response, status.Errorf(codes.InvalidArgument, "invalid format for grpc-previous-rpc-attempts header: %v", mdRetry[0])
+			}
+
+			if retry == attempt {
+				break
+			}
+
```

Comment:
Please don't include newlines at the end of every else-if block. The same applies when you switch over to `switch` as I've asked for in another comment.